### ISSUE TITLE
[BUG] Legacy join APIs return extraneous columns

### ISSUE THREAD

**Issue by jlowe** [rapidsai/cudf#7762]
**Describe the bug**
While updating the Java bindings to the new join gather map APIs added in #7454, I noticed that the old join APIs now return more columns than they did before.  They seem to be returning a table that contains the key columns from both the left and right tables, whereas before they returned only the key columns as they appeared in the left table (updated for the join result).

For example take two tables, the left having columns A,B and the right having columns C,D and we want to perform a join on A == C.  Previously the join APIs would return a table with 3 columns, A, B, and D, but now they return 4 columns.

**Steps/Code to reproduce bug**
Call the `left_join`, `inner_join`, and `full_join` APIs before and after #7454.

**Expected behavior**
Redundant key columns should not be manifested.  Manifesting the right key columns on an inner or left join seems to serve no purpose, as the left key columns already contain the proper answer for the join.

**Comment by harrism** [comment]
@shwina did you see this?

**Comment by shwina** [comment]
This was by design, although I can concede a lazy one on my part. @jrhemstad wanted the "common columns" parameter from the legacy APIs gone and I didn't replace it with reasonable default behaviour 

One solution is to get rid of these APIs entirely. Python (and I believe Spark) don't use them anymore. If we still want them for the convenience of libcudf users, we could make it such that only key columns from the left are preserved for inner and left joins.

**Comment by shwina** [comment]
Sorry for closing! Wrong button

**Comment by jlowe** [comment]
Still relevant.  The API should either be fixed or removed.

**Comment by jlowe** [comment]
I think this is still relevant, but low priority.

> One solution is to get rid of these APIs entirely.

I think this is the most appropriate solution, since I cannot see a valid reason callers would want or expect duplicated key columns in the join output.
